tests/integration/test_path_ingress_traefik.py

```diff
@@ -5,32 +5,41 @@
 import json
 from urllib.request import Request, urlopen
 
+import jubilant
+
 from . import TRAEFIK, ZINC, retry
-from .juju import Juju
 
 
-def test_deploy(zinc_charm, zinc_oci_image):
+def test_deploy(juju: jubilant.Juju, zinc_charm, zinc_oci_image):
     """Test that Zinc can be related with Traefik for ingress."""
     apps = [ZINC, TRAEFIK]
     traefik_config = {"routing_mode": "path", "external_hostname": "foo.bar"}
 
-    Juju.deploy(zinc_charm, alias=ZINC, resources={"zinc-image": zinc_oci_image})
-    Juju.deploy(TRAEFIK, config=traefik_config, trust=True)
-    Juju.integrate(*apps)
-    Juju.wait_for_idle(apps, timeout=1000)
+    juju.deploy(zinc_charm, app=ZINC, resource={"zinc-image": zinc_oci_image})
+    juju.deploy(TRAEFIK, config=traefik_config, trust=True)
+    juju.integrate(*apps)
+    juju.wait(jubilant.all_active)
 
 
-def test_ingress_setup():
+def test_ingress_setup(juju: jubilant.Juju):
     """Test that Zinc/Traefik are configured correctly."""
-    result = Juju.run(f"{TRAEFIK}/0", "show-proxied-endpoints")
-    j = json.loads(result["proxied-endpoints"])
+    result = juju.run(f"{TRAEFIK}/0", "show-proxied-endpoints")
+    j = json.loads(result.results["proxied-endpoints"])
+
+    if juju.model:
+        model_name = juju.model
+    else:
+        raise Exception("Juju model not found")
```

Comment:
I think the asserts you've used elsewhere are better/simpler:
```suggestion
    model_name = juju.model
    assert model_name is not None
```